packages/labextension/src/index.ts

```diff
@@ -2,34 +2,82 @@ import {
   JupyterFrontEnd,
   JupyterFrontEndPlugin,
 } from '@jupyterlab/application';
+import { INotebookTracker } from '@jupyterlab/notebook';
+import { LabIcon } from '@jupyterlab/ui-components';
+import { ICommandPalette } from '@jupyterlab/apputils';
+import { nullTranslator } from '@jupyterlab/translation';
+import { KernelUsagePanel } from './panel';
+import tachometer from '../style/tachometer.svg';
 
 import { IStatusBar } from '@jupyterlab/statusbar';
 
 import { ITranslator } from '@jupyterlab/translation';
 
 import { MemoryUsage } from './memoryUsage';
 
+namespace CommandIDs {
+  export const getKernelUsage = 'kernel-usage:get';
+}
+
 /**
  * Initialization data for the jupyter-resource-usage extension.
  */
 const extension: JupyterFrontEndPlugin<void> = {
-  id: '@jupyter-server/resource-usage:memory-status-item',
+  id: '@jupyter-server/resource-usage:memory-kernel-status-item',
   autoStart: true,
-  requires: [IStatusBar, ITranslator],
+  optional: [IStatusBar, ITranslator, ICommandPalette, INotebookTracker],
   activate: (
     app: JupyterFrontEnd,
-    statusBar: IStatusBar,
-    translator: ITranslator
+    statusBar: IStatusBar | null,
+    translator: ITranslator | null,
+    palette: ICommandPalette | null,
+    notebookTracker: INotebookTracker | null
   ) => {
+    translator = translator || nullTranslator;
+    const trans = translator.load('jupyterlab');
```

Comment:
I think we should extension-specific domain. https://jupyterlab.readthedocs.io/en/stable/extension/internationalization.html.

Also, on more general point, should these be two separate plugins so that they can be disabled easily? No strong opinions here, just suggesting.

Reply:
> I think we should extension-specific domain. https://jupyterlab.readthedocs.io/en/stable/extension/internationalization.html.

Done in https://github.com/jupyter-server/jupyter-resource-usage/pull/164/commits/f1a8bedd8a68e14e164eec7a7a35129f5524f47d.

> Also, on more general point, should these be two separate plugins so that they can be disabled easily? No strong opinions here, just suggesting.

No opinion, I will let others comment.

Reply:
> Also, on more general point, should these be two separate plugins so that they can be disabled easily? No strong opinions here, just suggesting.

Yes that was also one of the observations in https://github.com/jupyter-server/jupyter-resource-usage/pull/164#discussion_r1050648830.

Reply:
Please LMK if https://github.com/jupyter-server/jupyter-resource-usage/pull/164/commits/b97114320a2a86d0983a7fb70eea00ed12ff8da4 is fine.